lib/dotcom/trip_plan/itinerary_group.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Dotcom.TripPlan.ItineraryGroup do
+  @moduledoc """
+  A single group of related itineraries
+  """
+
+  alias Dotcom.TripPlan.Itinerary
+
+  defstruct [:itineraries, :representative_index, :representative_time, :summary]
+
+  @type summarized_leg :: %{
+          routes: [Routes.Route.t()],
+          walk_minutes: non_neg_integer()
+        }
+
+  @type summary :: %{
+          accessible?: boolean() | nil,
+          duration: non_neg_integer(),
+          start: DateTime.t(),
+          stop: DateTime.t(),
+          summarized_legs: [summarized_leg()],
+          tag: String.t(),
+          total_cost: non_neg_integer(),
+          walk_distance: float()
+        }
+
+  @type t :: %__MODULE__{
+          itineraries: [Itinerary.t()],
+          representative_index: non_neg_integer(),
+          representative_time: :start | :stop,
+          summary: summary()
+        }
+
+  @doc """
+  List of either start times or stop times for this group
+  """
+  @spec options_text(t()) :: String.t() | nil
+  def options_text(%__MODULE__{itineraries: []}), do: nil
+  def options_text(%__MODULE__{itineraries: [_single]}), do: nil
+
+  def options_text(
+        %__MODULE__{
+          representative_index: representative_index,
+          representative_time: start_or_stop
+        } =
+          group
+      ) do
+    {_, other_times} =
+      group
+      |> all_times()
+      |> List.pop_at(representative_index)
+
+    phrase = options_phrase(start_or_stop, Enum.count(other_times))
+
+    formatted_times =
+      other_times
+      |> Enum.map_join(", ", &Timex.format!(&1, "%-I:%M", :strftime))
+
+    "Similar #{phrase} #{formatted_times}"
+  end
+
+  defp options_phrase(:stop, 1), do: "trip arrives by"
+  defp options_phrase(:stop, _), do: "trips arrive by"
```

Comment:
Nit (non-blocking): "Arrive by" is a constraint, not a description, but (if I'm reading the code right), these strings are descriptive rather than constraining. 

What I mean is that even if the option that a user selected is "Arrive by", it still feels more natural when describing a trip to say "Similar trips arrive at 3:23, 3:53", rather than "Similar trips arrive by..."

```suggestion
  defp options_phrase(:stop, 1), do: "trip arrives at"
  defp options_phrase(:stop, _), do: "trips arrive at"
```